File: martianrun/core/src/com/gamestudio24/martianrun/actors/menu/ShareButton.java
package com.gamestudio24.martianrun.actors.menu;

import com.badlogic.gdx.math.Rectangle;
import com.gamestudio24.martianrun.utils.Constants;

public class ShareButton extends GameButton {

    private final ShareButtonListener listener;

    public ShareButton(Rectangle bounds, ShareButtonListener listener) {
        super(bounds);
        this.listener = listener;
    }

    @Override
    protected String getRegionName() {
        return Constants.SHARE_REGION_NAME;
    }

    @Override
    public void touched() {
        listener.onShare();
    }

    public interface ShareButtonListener {
        void onShare();
    }
}

File: martianrun/core/src/com/gamestudio24/martianrun/actors/menu/ShareButton.kt
package com.gamestudio24.martianrun.actors.menu

import com.badlogic.gdx.math.Rectangle
import com.gamestudio24.martianrun.utils.Constants

class ShareButton(bounds: Rectangle, private val listener: ShareButtonListener) : GameButton(bounds) {
    override fun getRegionName(): String? {
        return Constants.SHARE_REGION_NAME
    }

    override fun touched() {
        listener.onShare()
    }

    interface ShareButtonListener {
        fun onShare()
    }
}
